spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicConfigurationSupport.java

```diff
@@ -0,0 +1,485 @@
+	 * To configure it, consider overriding the {@link #configureRetryTopicConfigurer()}.
+	 * @param kafkaConsumerBackoffManager the global {@link KafkaConsumerBackoffManager}.
+	 * @param destinationTopicResolver the global {@link DestinationTopicResolver}.
+	 * @param beanFactory the {@link BeanFactory}.
+	 * @return the instance.
+	 * @see KafkaListenerAnnotationBeanPostProcessor
+	 */
+	@Bean(name = RetryTopicBeanNames.RETRY_TOPIC_CONFIGURER_BEAN_NAME)
+	public RetryTopicConfigurer retryTopicConfigurer(@Qualifier(KafkaListenerConfigUtils.KAFKA_CONSUMER_BACK_OFF_MANAGER_BEAN_NAME)
+																KafkaConsumerBackoffManager kafkaConsumerBackoffManager,
+													@Qualifier(RetryTopicBeanNames.DESTINATION_TOPIC_RESOLVER_BEAN_NAME)
+															DestinationTopicResolver destinationTopicResolver,
+													BeanFactory beanFactory) {
+
+		DestinationTopicProcessor destinationTopicProcessor = this.componentFactory
+				.destinationTopicProcessor(destinationTopicResolver);
+		DeadLetterPublishingRecovererFactory dlprf = this.componentFactory
+				.deadLetterPublishingRecovererFactory(destinationTopicResolver);
+		ListenerContainerFactoryConfigurer lcfc = this.componentFactory
+				.listenerContainerFactoryConfigurer(kafkaConsumerBackoffManager,
+						dlprf, this.componentFactory.internalRetryTopicClock());
+		ListenerContainerFactoryResolver factoryResolver = this.componentFactory
+				.listenerContainerFactoryResolver(beanFactory);
+		RetryTopicNamesProviderFactory retryTopicNamesProviderFactory =
+				this.componentFactory.retryTopicNamesProviderFactory();
+
+		processDeadLetterPublishingContainerFactory(dlprf);
+		processListenerContainerFactoryConfigurer(lcfc);
+
+		RetryTopicConfigurer retryTopicConfigurer = this.componentFactory
+				.retryTopicConfigurer(destinationTopicProcessor, lcfc,
+						factoryResolver, retryTopicNamesProviderFactory);
+
+		Consumer<RetryTopicConfigurer> configurerConsumer = configureRetryTopicConfigurer();
+		Assert.notNull(configurerConsumer, "configureRetryTopicConfigurer cannot return null.");
+		configurerConsumer.accept(retryTopicConfigurer);
+		return retryTopicConfigurer;
+	}
+
+	/**
+	 * Override this method if you need to configure the {@link RetryTopicConfigurer}.
+	 * @return a {@link RetryTopicConfigurer} consumer.
+	 */
+	protected Consumer<RetryTopicConfigurer> configureRetryTopicConfigurer() {
+		return retryTopicConfigurer -> {
+		};
+	}
+
+	/**
+	 * Internal method for processing the {@link DeadLetterPublishingRecovererFactory}.
+	 * Consider overriding the {@link #configureDeadLetterPublishingContainerFactory()}
+	 * method if further customization is required.
+	 * @param deadLetterPublishingRecovererFactory the instance.
+	 */
+	private void processDeadLetterPublishingContainerFactory(
+			DeadLetterPublishingRecovererFactory deadLetterPublishingRecovererFactory) {
+		CustomizersConfigurer customizersConfigurer = new CustomizersConfigurer();
```

Comment:
With multi-line signatures, we prefer a blank line before the start of the code.